Source/DafnyStandardLibraries/examples/FileIO/ReadFromFile.dfy

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT
+ *******************************************************************************/
+
+module ReadFromFile {
+  import Std.FileIO
+
+  @Test
+  method Test() {
+    // TODO: extern function for the expected error prefix
+    theMain("examples/FileIO/../FileIO/data.txt", "");
+  }
+
+  method theMain(dataPath: string, expectedErrorPrefix: string) {
+
+      // Happy path: read from the data file, and check that we see the expected content.
+    {
+      var expectedStr := "Hello world\nGoodbye\n";
+      var res := FileIO.ReadUTF8FromFile(dataPath);
+      expect res.Success?, "unexpected failure: " + res.error;
+      expect res.value == expectedStr;
+    }
+
+      // Failure path: attempting to read from a blank file path should never work.
```

Comment:
A second failure case where the bytes are not UTF8 would be very valuable too, especially since AFAICT we don't prove that the index is correct. Could also put it in UnicodeExamples.dfy

Reply:
UnicodeExamples already has a test for ill cases, so I just added 2 expect lines to demonstrate what to expect as an error message.